src/nwb_benchmarks/core/_network_tracker.py

```diff
@@ -0,0 +1,59 @@
+"""
+Network tracker for capturing network traffic and performance metrics during the execution of particular
+methods or code snippets.
+"""
+
+import contextlib
+import os
+import pathlib
+import time
+from typing import Union
+
+from ._capture_connections import CaptureConnections
+from ._network_profiler import NetworkProfiler
+from ._network_statistics import NetworkStatistics
+
+
+@contextlib.contextmanager
+def network_activity_tracker(tshark_path: Union[pathlib.Path, None] = None):
+    network_tracker = NetworkTracker()
+
+    try:
+        network_tracker.start_network_capture(tshark_path=tshark_path)
+        time.sleep(0.3)
+
+        t0 = time.time()
+        yield network_tracker
+    finally:
+        network_tracker.stop_network_capture()
+
+        t1 = time.time()
+        network_total_time = t1 - t0
+        network_tracker.network_statistics["network_total_time"] = network_total_time
+
+
+class NetworkTracker:
+    def start_network_capture(self, tshark_path: Union[pathlib.Path, None] = None):
```

Comment:
```suggestion
class NetworkTracker:
    """
    Track network traffic and compute network statistics for operations performed by the current 
    Python process during the active period of the tracker, which is started by `start_network_capture`
    and ended by `stop_network_capture`.
    
    **Instance variables created by  `start_network_capture`**
    * `self.connections_thread` : Instance of `CaptureConnections` used to relate network connections to process IDs
    * `self.network_profiler`: Instance of `NetworkProfiler` used to capture network traffic with TShark
    
     **Instance variables created by  `stop_network_capture`**
     * `self.pid_connections` : connections for the PID of this process 
     * `self.pid_packets` : Network packets associated with this PID (i.e., `os.getpid()`)
     * `self.network_statistics` : Network statistics calculated via `NetworkStatistics.get_statistics`
     * `self.asv_network_statistics` : The network statistics wrapped in a dict for compliance with ASV
    
    """
    def start_network_capture(self, tshark_path: Union[pathlib.Path, None] = None):
```

Reply:
Should we add a ``__init__`` here that sets all these instance variables to `None`?

Reply:
I'm impartial to that; all up to you